cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/store/AppMetadataStoreTest.java

```diff
@@ -133,57 +136,193 @@ public void apply() throws Exception {
     });
   }
 
-  @Test
-  public void testSmallerSourceIdRecords() throws Exception {
-    DatasetId storeTable = NamespaceId.DEFAULT.dataset("testRandomSourceIdRecords");
+  private AppMetadataStore getMetadataStore(String tableName) throws Exception {
+    DatasetId storeTable = NamespaceId.DEFAULT.dataset(tableName);
     datasetFramework.addInstance(Table.class.getName(), storeTable, DatasetProperties.EMPTY);
 
     Table table = datasetFramework.getDataset(storeTable, ImmutableMap.<String, String>of(), null);
     Assert.assertNotNull(table);
-    final AppMetadataStore metadataStoreDataset = new AppMetadataStore(table, cConf, new AtomicBoolean(false));
-    TransactionExecutor txnl = txExecutorFactory.createExecutor(
-      Collections.singleton((TransactionAware) metadataStoreDataset));
+    return new AppMetadataStore(table, cConf, new AtomicBoolean(false));
+  }
+
+  private TransactionExecutor getTxExecutor(AppMetadataStore metadataStoreDataset) {
+    return txExecutorFactory.createExecutor(Collections.singleton((TransactionAware) metadataStoreDataset));
+  }
 
+  @Test
+  public void testSmallerSourceIdRecords() throws Exception {
+    final AppMetadataStore metadataStoreDataset = getMetadataStore("testSmallerSourceIdRecords");
+    TransactionExecutor txnl = getTxExecutor(metadataStoreDataset);
+    // STARTING status is persisted with the largest sourceId
```

Comment:
I wasn't able to tell what a 'sourceId' is and what it's used for from the javadoc for AppMetadataStore. Please add additional javadocs to the AppMetadataStore methods, otherwise someone who is not familiar with the code cannot tell what it does and what this is testing.